torchreid/losses/am_softmax.py

```diff
@@ -205,6 +206,15 @@ def forward(self, cos_theta, target, iteration=None):
             else:
                 losses = F.cross_entropy(output, target, reduction='none')
 
+            if (self.aug_type and aug_index is not None and lam is not None):
+                targets1 = torch.zeros(output.size()).scatter_(1, target.unsqueeze(1).data.cpu(), 1)
+                if self.use_gpu:
+                    targets1 = targets1.cuda()
+
+                targets2 = targets1[aug_index]
+                new_targets = targets1 * lam + targets2 * (1 - lam)
+                losses = (- new_targets * F.log_softmax(output, dim=1)).sum(dim=1)
```

Comment:
Seems like tat's not completely correct since `output` is generated without knowing new_targets.
Can the argmax in new_targets and targets1 be different?

Reply:
Fix this
Please, check that everything is correct now